[mtt] mlx5dv_devx_qp_modify(503) failed, syndrome 0: Invalid argument when UCX_IB_SL=2

Configuration:
```
OMPI: 4.0.3rc4
MOFED: MLNX_OFED_LINUX-5.0-1.0.0.0
Module: hpcx-gcc (2020-04-06)
Test module: mtt-tests/hpcx-gcc
Nodes: hercules x96 (ppn=32(x96), nodelist=clx-hercules-[019-025,027,030-031,033-118])
```

MTT log: http://e2e-gw.mellanox.com:4080/mnt/lustre/users/mtt/scratch/ucx_ompi/20200406_092051_8583_576753_clx-hercules-019/html/test_stdout_rtymMk.txt

Only 3 nodes are required to reproduce.
It doesn't reproduce with `UCX_IB_SL=0` (and without `UCX_IB_SL` explicitly set).

Cmd:
`mpirun -np 3 -mca btl self --tag-output --timestamp-output -mca pml ucx -mca coll '^hcoll' -x UCX_ERROR_SIGNALS=SIGILL,SIGSEGV,SIGBUS,SIGFPE,SIG
ABRT -x UCX_HANDLE_ERRORS=bt -x UCX_IB_PCI_RELAXED_ORDERING=off --bind-to core -x UCX_NET_DEVICES=mlx5_2:1,mlx5_3:1 -mca osc ucx -x UCX_IB_REG_ME
THODS=rcache,direct -x UCX_TLS=dc_x,self,sm -x UCX_IB_SL=2 -mca pmix_base_async_modex 0 -mca mpi_add_procs_cutoff 100000 -mca pmix_base_collect_d
ata 1 --map-by node /hpc/local/benchmarks/daily/next/2020-04-06/mtt-tests-gcc/installs/_WAn/tests/mpich_tests/mtt-tests.git/mpich/test/mpi/pt2pt/
large_message`

Output:
```sh
Mon Apr  6 14:58:07 2020[1,1]<stdout>:[1586174287.331106] [clx-hercules-015:32494:0]     ib_mlx5_dv.c:217  UCX  ERROR mlx5dv_devx_qp_modify(503) failed, syndrome 0: Invalid argument
Mon Apr  6 14:58:07 2020[1,2]<stdout>:[1586174287.332488] [clx-hercules-016:31515:0]     ib_mlx5_dv.c:217  UCX  ERROR mlx5dv_devx_qp_modify(503) failed, syndrome 0: Invalid argument
Mon Apr  6 14:58:07 2020[1,0]<stdout>:[1586174287.339570] [clx-hercules-009:18163:0]     ib_mlx5_dv.c:217  UCX  ERROR mlx5dv_devx_qp_modify(503) failed, syndrome 0: Invalid argument
Mon Apr  6 14:58:07 2020[1,1]<stderr>:[clx-hercules-015:32494] pml_ucx.c:273  Error: Failed to create UCP worker
Mon Apr  6 14:58:07 2020[1,2]<stderr>:[clx-hercules-016:31515] pml_ucx.c:273  Error: Failed to create UCP worker
--------------------------------------------------------------------------
No components were able to be opened in the pml framework.

This typically means that either no components of this type were
installed, or none of the installed components can be loaded.
Sometimes this means that shared libraries required by these
components are unable to be found/loaded.

  Host:      clx-hercules-015
  Framework: pml
--------------------------------------------------------------------------
Mon Apr  6 14:58:07 2020[1,1]<stderr>:[clx-hercules-015:32494] PML ucx cannot be selected
Mon Apr  6 14:58:07 2020[1,2]<stderr>:[clx-hercules-016:31515] PML ucx cannot be selected
Mon Apr  6 14:58:09 2020[1,0]<stderr>:[clx-hercules-009:18163] pml_ucx.c:273  Error: Failed to create UCP worker
------------------------------------------------------------
A process or daemon was unable to complete a TCP connection
to another process:
  Local host:    mngx-orion-01
  Remote host:   clx-hercules-015
This is usually caused by a firewall on the remote host. Please
check that any firewall (e.g., iptables) has been disabled and
try again.
------------------------------------------------------------
[mngx-orion-01:28954] 2 more processes have sent help message help-mca-base.txt / find-available:none found
[mngx-orion-01:28954] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages
```

--- a/src/uct/ib/dc/dc_mlx5_devx.c
+++ b/src/uct/ib/dc/dc_mlx5_devx.c
@@ -109,7 +109,13 @@ uct_dc_mlx5_iface_devx_dci_connect(uct_dc_mlx5_iface_t *iface,
     UCT_IB_MLX5DV_SET(qpc, qpc, log_msg_max, UCT_IB_MLX5_LOG_MAX_MSG_SIZE);
     UCT_IB_MLX5DV_SET(qpc, qpc, atomic_mode, UCT_IB_MLX5_ATOMIC_MODE);
     UCT_IB_MLX5DV_SET(qpc, qpc, rae, true);
-    UCT_IB_MLX5DV_SET(qpc, qpc, primary_address_path.sl, iface->super.super.super.config.sl);
+    if (uct_ib_iface_is_roce(&iface->super.super.super)) {
+        UCT_IB_MLX5DV_SET(qpc, qpc, primary_address_path.eth_prio,
+                          iface->super.super.super.config.sl);
+    } else {
+        UCT_IB_MLX5DV_SET(qpc, qpc, primary_address_path.sl,
+                          iface->super.super.super.config.sl);
+    }
 
     status = uct_ib_mlx5_devx_modify_qp(qp, in_2rtr, sizeof(in_2rtr),
                                         out_2rtr, sizeof(out_2rtr));

